core-container/entry.sh

```diff
@@ -16,88 +16,487 @@
+    echo Started.
+
+    # Waiting to be ready
+    echo "Waiting for ${KUBECONFIG}..."
+    while [ ! -f "${KUBECONFIG}" ]; do sleep 5; done
+    echo 'Waiting for "root:compute" workspace...'
+    until [ "$(kubectl ws root:compute 2> /dev/null)" != "" ]; do sleep 5;done
+    echo '"root:compute" workspace is ready'.
+    echo "kcp version: $(kubectl version --short 2> /dev/null | grep kcp | sed 's/.*kcp-//')"
+    kubectl ws root
+
+    # Generate the external.kubeconfig
+    if [ -n "$EXTERNAL_HOSTNAME" ] && [ ! -d "${PWD}/.kcp-${EXTERNAL_HOSTNAME}" ]; then
+        echo Creating external.kubeconfig...
+        switch-domain .kcp/admin.kubeconfig .kcp/external.kubeconfig root ${EXTERNAL_HOSTNAME} ${EXTERNAL_PORT} ${pieces_external[0]}
+        switch-domain .kcp/admin.kubeconfig .kcp/cluster.kubeconfig root ${KUBESTELLAR_SERVICE} 6443 ${pieces_cluster[0]}
+    fi
+
+    # Ensure kubeconfig secret
+    echo Creating the kubestellar secret...
+    (
+        KUBECONFIG=
+        if [ -n "${EXTERNAL_HOSTNAME}" ]; then
+            kubectl create secret generic kubestellar --from-file="${PWD}/.kcp/admin.kubeconfig" --from-file="${PWD}/.kcp/cluster.kubeconfig" --from-file="${PWD}/.kcp/external.kubeconfig"
+        else
+            kubectl create secret generic kubestellar --from-file="${PWD}/.kcp/admin.kubeconfig"
+        fi
+    )
+    echo "***READY***"
+    sleep infinity
+}
+
+
+function run_init() {
+    echo "--< Starting init >--"
+    wait_kcp_ready
+    kubestellar init --local-kcp false
+    # ensure_root_compute_configd
+    # ensure_espw
+    kubectl ws root
+    ensure_invs $ENSURE_IMW
+    ensure_wmws $ENSURE_WMW
+    echo "***READY***"
+    sleep infinity
+}
+
+
+function run_mailbox_controller() {
+    echo "--< Starting mailbox-controller >--"
+    wait-kubestellar-ready
+    kubectl ws root:espw
+    if ! mailbox-controller -v=${VERBOSITY} ; then
+        echoerr "unable to start mailbox-controller!"
+        exit 1
+    fi
+}
+
+
+function run_where_resolver() {
+    echo "--< Starting where-resolver >--"
+    wait-kubestellar-ready
+    kubectl ws root:espw
+    if ! kubestellar-where-resolver -v ${VERBOSITY} ; then
+        echoerr "unable to start kubestellar-where-resolver!"
+        exit 1
     fi
+}
+
+
+function run_placement_translator() {
+    echo "--< Starting placement-translator >--"
+    wait-kubestellar-ready
+    kubectl ws root:espw
+    if ! placement-translator --allclusters-context  "system:admin" -v=${VERBOSITY} ; then
+        echoerr "unable to start mailbox-controller!"
+        exit 1
+    fi
+}
+
+
+echo "--< Starting KubeStellar container >--"
+
+echo "Environment variables:"
+if [ $# -ne 0 ] ; then
+    ACTION="$1"
+else
+    ACTION="sleep"
+fi
+echo "ACTION=${ACTION}"
+if [ "$VERBOSITY" == "" ]; then
+    VERBOSITY="2"
 fi
+if [ "$ESPW_NAME" == "" ]; then
+    ESPW_NAME="espw"
+fi
+echo "ESPW_NAME=${ESPW_NAME}"
+echo "VERBOSITY=${VERBOSITY}"
+if [ "$ENSURE_IMW" == "" ]; then
+    ENSURE_IMW=""
```

Comment:
What's the point of changing the value from empty string to empty string?

Reply:
ops